lint/src/main/java/org/datacommons/lint/Lint.java

```diff
@@ -0,0 +1,162 @@
+
+  @CommandLine.Parameters(
+      arity = "1..*",
+      description =
+          ("List of input files. This depends on the --format option. "
+              + "For 'tmcf', the files are expected to be "
+              + "Template MCF files. For 'mcf', the files are expected to be Instance MCF "
+              + "files. For tmcfCsv, the first file should be a Template MCF file, followed "
+              + "by one or more CSVs compatible with the provided TMCF file."))
+  private File[] files;
+
+  enum FormatType {
+    mcf,
+    tmcf,
+    tmcfCsv
+  };
+
+  @CommandLine.Option(
+      names = {"-f", "--format"},
+      description = "Format of input files: ${COMPLETION-CANDIDATES}",
+      required = true)
+  private FormatType formatType;
+
+  enum CommandType {
+    chk,
+    genMcf
+  };
+
+  @CommandLine.Option(
+      names = {"-c", "--cmd"},
+      description =
+          "The command to run: ${COMPLETION-CANDIDATES}.  'chk' runs "
+              + "syntax checks on the input files.  'genMcf' produces instance MCF output, "
+              + "and only makes sense when --format is 'tmcfCsv'. Default: 'chk'",
+      defaultValue = "chk",
+      required = true)
+  private CommandType cmdType;
+
+  @CommandLine.Option(
+      names = {"-d", "--delimiter"},
+      defaultValue = ",",
+      description =
+          "Delimiter of the input CSV file. Relevant only when --format is 'tmcfCsv'. "
+              + "Default: ',' (comma)")
+  private char delimiter;
+
+  @CommandLine.Option(
+      names = {"-o", "--outputDir"},
+      description = "Directory to write output files. Default: /tmp",
+      defaultValue = "/tmp",
+      scope = CommandLine.ScopeType.INHERIT)
+  private File outputDir;
+
+  @CommandLine.Option(
+      names = {"-r", "--isResolved"},
+      defaultValue = "false",
+      description =
+          "Indicates whether the input file is resolved. Relevant only when "
+              + "--format is "
+              + "'mcf'. Default: false")
+  private boolean isResolved;
+
+  @CommandLine.Spec CommandLine.Model.CommandSpec spec; // injected by picocli
+
+  public static void main(String... args) {
+    System.exit(new CommandLine(new Lint()).execute(args));
+  }
+
+  @Override
+  public Integer call() throws IOException {
+    if (formatType == FormatType.tmcfCsv) {
+      if (files.length < 2) {
+        throw new CommandLine.ParameterException(
+            spec.commandLine(), "At least two input files required with option 'tmcfCsv'");
+      }
+      return HandleTableFormat();
+    } else {
+      if (cmdType == CommandType.genMcf) {
+        throw new CommandLine.ParameterException(
+            spec.commandLine(), "Command 'genMcf' is only applicable with --format 'tmcfCsv'");
+      }
+      return HandleNodeFormat();
+    }
+  }
+
+  private Integer HandleNodeFormat() throws IOException {
+    Mcf.McfType mcf_type =
+        (formatType == FormatType.mcf ? Mcf.McfType.INSTANCE_MCF : Mcf.McfType.TEMPLATE_MCF);
```

Comment:
if this is inferred from filetype, we can do a simple check to see if there are any column references in the tmcf files first.

Reply:
This is obsolete now.